drivers/usb/device/usb_dc_usbip.c

```diff
@@ -0,0 +1,922 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <zephyr.h>
+#include <kernel.h>
+#include <sys/byteorder.h>
+#include <usb/usb_common.h>
+#include <usb/usbstruct.h>
+#include <usb/usb_device.h>
+#include <drivers/usb/usb_dc.h>
+#include <net/socket.h>
+
+#include "usb_dc_usbip.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(usbip, CONFIG_USB_DRIVER_LOG_LEVEL);
+
+extern struct usb_desc_header __usb_descriptor_start[];
+
+#define USBIP_IN_EP_NUM		8
+#define USBIP_OUT_EP_NUM	8
+#define USBIP_MAX_PACKET_SIZE	64
+
+K_KERNEL_STACK_MEMBER(usbip_thread_stack, 2048);
+
+static struct k_thread usbip_thread;
+static bool attached;
+
+struct usb_ep_ctrl_prv {
+	uint8_t ep_ena;
+	uint16_t mps;
+	usb_dc_ep_callback cb;
+	uint32_t data_len;
+	uint8_t buf[64];
+	uint8_t buf_len;
+};
+
+static struct usbip_ctrl_prv {
+	usb_dc_status_callback status_cb;
+	struct usb_ep_ctrl_prv in_ep_ctrl[USBIP_IN_EP_NUM];
+	struct usb_ep_ctrl_prv out_ep_ctrl[USBIP_OUT_EP_NUM];
+	int connfd;
+	int seqnum;
+	int devid;
+	uint32_t cmd_num_in;
+	uint32_t cmd_num_out;
+	uint8_t attached;
```

Comment:
perhaps bool here or if you want to have small memory usage make it a bitfield, then the functions setting the value could just use true/false. End result is the same but it is more clearer if using a bool value.